src/DataImport/gb_parse.py

```diff
@@ -67,23 +70,105 @@ def parse_genbank(genbank_file):
                         feature_type = 'rRNA'
 
                 # grabs DNA sequence from record object
-                dna_seq = str(feature.extract(contig))
-
+                dna_seq = str(feature.extract(contig).seq)
                 record_list.append({
-                    'type':feature_type,
-                    'dna_seq':dna_seq,
-                    'aa_seq':aa_seq,
-                    'locus_tag':locus_tag,
-                    'annotation':annotation,
-                    'location':{
-                        'start':int(feature.location.start),
-                        'end':int(feature.location.end),
-                        'strand':feature.location.strand,
-                        'contig':contig.id},
+                    'type': feature_type,
+                    'dna_seq': dna_seq,
+                    'aa_seq': aa_seq,
+                    'locus_tag': locus_tag,
+                    'annotation': annotation,
+                    'location': {
+                        'start': int(feature.location.start),
+                        'end': int(feature.location.end),
+                        'strand': feature.location.strand,
+                        'contig': contig.id},
                     'species':species
                 })
     return record_list
 
+
+def parse_genbank_and_insert(genbank_file, collection):
```

Comment:
I think this is a great example of where to break up functions. Making both of these big for loops into their own functions would be beneficial. A general rule of thumb is if you have to use the word "and" when describing what the function does it should be split up.